lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -37,7 +37,7 @@ def self.retrieve_new_ids(last_id)
       end
 
       # Stop if all the remaining results were in this response - we're done. Otherwise, keep repeating.
-      break if results_in_response == total_results
+      break if (results_in_response == total_results) || CDO.env == 'development'
```

Comment:
what's the reason for breaking in the development env?

Reply:
Just added a comment :) Basically when called from [here](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/contact_rollups/v2/contact_rollups.rb), it'll download all Pardot contacts (which was 30K for me this afternoon). This is a hack to limit to only download the first iteration through this loop. Probably not the cleanest approach...

Reply:
Thinking about this more, this change doesn't seem to be related to the rest of this PR. It should be in its own PR.

In addition, having implicit restriction/configuration inside a function could cause problem later. A better option could be making the the limit explicit by adding it to function inputs `ContactRollupsPardotMemory.add_and_update_pardot_ids(last_id, limit)` and `PardotV2.retrieve_new_ids(last_id, limit)`. The value of `limit` still depends on environment and is set in `ContactRollupsV2.build_contact_rollups`.

Reply:
Fair enough, I'll separate it out 👍